rpcs3/Emu/System.cpp

```diff
@@ -1865,6 +1867,121 @@ void Emulator::Stop(bool restart)
 	}
 }
 
+std::string Emulator::FormatTitle(double fps) const
+{
+	// Get version by substringing VersionNumber-buildnumber-commithash to get just the part before the dash
+	std::string version = rpcs3::get_version().to_string();
+
+	version = version.substr(0 , version.find_last_of('-'));
+
+	// Add branch and commit hash to version on frame unless it's master.
+	if ((rpcs3::get_branch().compare("master") != 0) && (rpcs3::get_branch().compare("HEAD") != 0))
+	{
+		version = version + "-" + rpcs3::get_version().to_string().substr((rpcs3::get_version().to_string().find_last_of('-') + 1), 8) + "-" + rpcs3::get_branch();
+	}
+
+	auto [title_format, life1] = g_cfg.misc.title_format.get();
+
+	// Parse title format string
+	std::string title_string;
+
+	// Backward compatibility hack
+	std::size_t fmt_start = 0;
+
+	if (g_cfg.misc.show_fps_in_title.get() == false)
+	{
+		if (title_format.starts_with("FPS: %F | "))
```

Comment:
This is a bad idea, the whole string should be hardcoded to be included inside %F.

Reply:
We could provide choice of 3 options: default, default with no FPS counter and custom. Then we could get rid of the whole show fps option.

Reply:
Why it's bad idea? You either get "FPS: Disabled" or remove the part of string at the beginning.

Reply:
I doubt anyone would mess with in such a way that it works wrong.

Reply:
I mean, the situation when this might fail is extremely artificial to begin with.

Reply:
or just.. display 0.00 when its disabled,
or other placeholder.